cores/esp32/Esp.cpp

```diff
@@ -199,10 +204,10 @@ bool EspClass::flashEraseSector(uint32_t sector)
 
 bool EspClass::flashWrite(uint32_t offset, uint32_t *data, size_t size)
 {
-    return spi_flash_write(offset, (uint32_t*) data, size) == ESP_OK;
+    return spi_flash_write_encrypted(offset, (uint32_t*) data, size) == ESP_OK;
 }
 
 bool EspClass::flashRead(uint32_t offset, uint32_t *data, size_t size)
 {
-    return spi_flash_read(offset, (uint32_t*) data, size) == ESP_OK;
+    return spi_flash_read_encrypted(offset, (uint32_t*) data, size) == ESP_OK;
```

Comment:
Same here, with flash encryption enabled, this will always try to decrypt data, even if it wasn't encrypted in the first place. This isn't a problem if data is always encrypted, but as I have mentioned above, unconditionally encrypting data may be problematic due to alignment restrictions and bit flipping not being possible.